drivers/stepper_control/zephyr_stepper_motion_control/stepper_motion_control_timing_source.h

```diff
@@ -7,32 +7,55 @@
 #define ZEPHYR_DRIVER_STEPPER_STEP_DIR_STEPPER_TIMING_SOURCE_H_
 
 #include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/counter.h>
+
+struct timing_source_config {
+	const struct device *counter;
+};
+
+struct timing_source_data {
+	const struct device *motion_control_dev;
+	uint64_t microstep_interval_ns;
+	struct k_work_delayable stepper_dwork;
+
+#ifdef CONFIG_ZEPHYR_STEPPER_MOTION_CONTROL_COUNTER_TIMING
+	struct counter_top_cfg counter_top_cfg;
+	bool counter_running;
+#endif /* CONFIG_ZEPHYR_STEPPER_MOTION_CONTROL_COUNTER_TIMING */
+};
```

Comment:
At the moment counter specific data is guarded by `CONFIG_ZEPHYR_STEPPER_MOTION_CONTROL_COUNTER_TIMING` while counter device is not. 

I think we should have separate structs for config and data per timing source type. This way we keep this interface clean of specific implementations and also allow other types of timing sources in the future without bloating the generic structs.

```C
struct stepper_timing_source_counter_config
{
	const struct device *counter;
};

struct stepper_timing_source_counter_data
{
	struct counter_top_cfg counter_top_cfg;
};

struct stepper_timing_source_work_cfg
{
};

struct stepper_timing_source_work_data
{
	struct k_work_delayable stepper_dwork;
};
```